app/cdap/components/WranglerGrid/NestedMenu/index.tsx

```diff
@@ -0,0 +1,156 @@
+      if (menuItem.hasOwnProperty('options') && menuItem?.options?.length > 0) {
+        let referenceIndex = -1;
+        menuComponentOptions.forEach((menuOptions, menuOptionsIndex) => {
+          menuOptions.forEach((eachOption, optionsIndex) => {
+            if (eachOption.value === menuItem.value) {
+              referenceIndex = menuOptionsIndex; // Getting index of one level higher
+            }
+          });
+        });
+        if (referenceIndex >= 0 && !anchorElement.includes(event.currentTarget)) {
+          // if array of anchor element does not have the current item clicked then we will insert it
+          const insertPosition = referenceIndex + 2; // +2 because 0, when clicked on icon, 1 when clicked on parent menu item, it means now whatever will come at position 2
+          anchorElement.splice(insertPosition, 0, event.currentTarget);
+          const updatedAnchors = anchorElement.slice(0, insertPosition + 1);
+          menuComponentOptions.splice(updatedAnchors.length - 2, 0, menuItem?.options);
+          setMenuComponentOptions((prev) =>
+            prev.length ? menuComponentOptions : [menuItem?.options]
+          );
+          setAnchorElement(updatedAnchors);
+        } else if (anchorElement.includes(event.currentTarget)) {
+          // if array of anchor element have current item then we have to remove it and reposition it to it's parent
+          const currentTargetIndex = findIndex(
+            anchorElement,
+            (anchor) => anchor == event.currentTarget
+          );
+          menuComponentOptions.splice(currentTargetIndex, 0, menuItem?.options);
+          setAnchorElement((prev) => prev.slice(0, currentTargetIndex + 1));
+          setMenuComponentOptions((prev) => menuComponentOptions.slice(0, currentTargetIndex + 1));
+        } else {
+          setAnchorElement((prev) => [...prev, event.currentTarget]);
+          setMenuComponentOptions((prev) =>
+            prev.length ? [...prev, menuItem?.options] : [menuItem?.options]
+          );
+        }
+      } else {
+        submitMenuOption(menuItem.value, menuItem.supportedDataType);
+        setAnchorElement(null);
+        menuToggleHandler(title);
+      }
+    }
+  };
+  return (
+    <>
+      <NestedMenuComponent
+        id="parent-menu"
+        data-testid="nested-menu-parent-root"
+        keepMounted
+        anchorEl={anchorElement?.length ? anchorElement[0] : null}
+        open={anchorElement?.length ? true : false}
+        getContentAnchorEl={null}
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+        transformOrigin={{ vertical: 'top', horizontal: 'left' }}
+        onClose={() => menuToggleHandler()}
+        onClick={(clickEvent) => {
+          clickEvent.preventDefault();
+          clickEvent.stopPropagation();
+        }}
+      >
+        {menuOptions?.map((eachOption, optionsIndex) => {
+          return (
+            <MenuItemComponent
```

Comment:
Does this require a `key`?

Reply:
Yes, it is required as we are mapping and maintaining the uniqueness.